api/src/OpenTelemetry/Attributes/Key.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+
+{- |
+Module      :  OpenTelemetry.Attributes.Attribute
+Copyright   :  (c) Ian Duncan, 2021
+License     :  BSD-3
+Description :  Key-value pair metadata used in 'OpenTelemetry.Trace.Span's, 'OpenTelemetry.Trace.Link's, and 'OpenTelemetry.Trace.Event's
+Maintainer  :  Ian Duncan
+Stability   :  experimental
+Portability :  non-portable (GHC extensions)
+-}
+module OpenTelemetry.Attributes.Key (
+  Key (..),
+  forget,
+) where
+
+import Data.String (IsString (..))
+import Data.Text (Text)
+import qualified Data.Text as T
+import GHC.Generics (Generic)
+import OpenTelemetry.Attributes.Attribute (Attribute)
+
+
+newtype Key a = Key {unkey :: Text} deriving stock (Show, Eq, Ord, Generic)
```

Comment:
```suggestion
-- | A 'Key' is a name for an attribute. The type parameter sets the type of the
-- attribute the key should be associated with. This is useful for standardising 
-- attribute keys, since we can define both the key and the type of value it is 
-- intended to record.
--
-- For example, we might define:
-- @  
--     -- See https://opentelemetry.io/docs/specs/semconv/attributes-registry/server/
--     serverPortKey :: Key Int
--     serverPortKey = "server.port"
```

Reply:
Having written this, I wonder if we shouldn't just call this `AttributeName`. It's more verbose, but perhaps better to be more descriptive?

Reply:
I think the general naming convention in the package has been to go for longer names that are non-ambiguous by themselves. e.g. we have `SpanProcessor` rather than just `Span.Processor`. So I think "AttributeKey" fits that approach better.

Reply:
Thanks for checking the spec for “key” vs. “name”. I prefer “key” for maps. I follow this project's customs although I prefer non-ambiguous name.

So I will rename this to `AttributeKey`.